nevergrad/functions/functionlib.py

```diff
@@ -62,6 +64,53 @@ def process(self, data: ArrayLike, deterministic: bool = True) -> np.ndarray:  #
     def _short_repr(self) -> str:
         return "Photonics"
 
+    
+# Example of ML problem.
+def _decision_tree_parametrization(depth: int):
+    # 10-folds cross-validation
+    num_data: int = 80
+    result: float = 0.
+    for cv in range(10):
+        list_X = []
+        list_X_test = []
+        for i in range(num_data):
+            if i % 10 != cv:
+                list_X += [[float(i)]]
+            else:
+                list_X_test += [[float(i)]]
+
+        # Convert to numpy arrays.
+        X = np.array([np.array(xi) for xi in list_X])
+        X_test = np.array([np.array(xi) for xi in list_X_test])
+        y = np.sin(X).ravel()
+
+        assert isinstance(depth, int), f"depth has class {type(depth)} and value {depth}."
+
+        # Fit regression model
+        regr = DecisionTreeRegressor(max_depth=depth)
+        regr.fit(np.asarray(X), np.asarray(y))
+
+        # Predict
+        pred_test = regr.predict(X_test)
+        y_test = np.sin(np.asarray(X_test)).ravel()
+        result += np.sum((y_test - pred_test)**2)
+    return result / num_data
+
+
+class MLTuning(ExperimentFunction):
+    """Class for generating ML hyperparameter tuning problems.
+    """
+    def __init__(self, problem_type: str):
+        self.name = problem_type
+        self._parameters = {x: y for x, y in locals().items() if x not in ["__class__", "self"]}
+
+        if problem_type == "1d_decision_tree_regression":
+            parametrization = p.Instrumentation(depth=p.Scalar(lower=1, upper=1200).set_integer_casting())        
+            super().__init__(_decision_tree_parametrization, parametrization)
+            self.register_initialization(**self._parameters)
```

Comment:
you can move this line out of the if-condition. Also, you should add descriptors if you plan on adding other problem types